lz4/frame/_frame.c

```diff
@@ -171,12 +170,13 @@ py_lz4f_makePrefs (PyObject * Py_UNUSED (self), PyObject * args,
     }
 
   prefs = calloc (1, sizeof (LZ4F_preferences_t));
-  frameInfo = (LZ4F_frameInfo_t)
   {
-    blkID, blkMode, chkSumFlag, 0, 0,
-    {0, 0}
-  };
-  prefs->frameInfo = frameInfo;
+    LZ4F_frameInfo_t frameInfo = {
+      blkID, blkMode, chkSumFlag, 0, 0,
+      {0, 0}
+    };
+    prefs->frameInfo = frameInfo;
+  }
   prefs->autoFlush = autoFlush;
   result = PyCapsule_New (prefs, NULL, NULL);
 
```

Comment:
This troubles me: the compound literal (which, actually I hate to see used and welcome expunging of!) assured static storage for frameInfo, so no problem with it persisting outside the function scope. It seems to me the alternative you've got here makes no such guarantee.

I think a better fix would be to allocate frameInfo explicitly (with a malloc).

Also, we should be checking the return value of calloc for prefs - I've been adding a lot of those to the frame code, but this is one I've missed.